modules/govuk_postgresql/manifests/db_admin_common_config.pp

```diff
@@ -0,0 +1,79 @@
+# == Class: govuk_postgresql::db_admin_common_config
+#
+# This machine class defines the common configuration used to
+# administer the app-specific DB-admin machines for PostgreSQL
+# RDS instances.
+#
+# === Parameters
+#
+# [*postgres_host*]
+#   Hostname of the RDS database to use.
+#   Default: undef
+#
+# [*postgres_user*]
+#   The PostgreSQL user to use for admisistering the database.
+#   Default: undef
+#
+# [*postgres_password*]
+#   The password corresponding to the above `postgres_user`.
+#   Default: undef
+#
+# [*postgres_port*]
+#   The port with which to connect to the `postgres_host`.
+#   Default: '5432'
+#
+class govuk_postgresql::db_admin_common_config(
```

Comment:
I don't think this is the right place for this class, since govuk_postgresql is for installing postgresql (as I recall).

I'd suggest perhaps creating a directory `modules/govuk/manifests/nodes` (similar to `modules/govuk/manifests/apps`) for specific node convenience. 

I'd have thought we'd want to be able to import this file into db_admin too - since it has postgres. If there are reason that stop that it shouldn't be in this file (in my opinion).

(great idea by the way - quick drive by review)

Reply:
Thanks, I've renamed the class, and, when I realised the Postgres config in the main db_admin node is the same as the per-app configs, I sided towards making the same change there too.

I tried your suggestion from Slack to use inheritance, e.g.

```
# == Class: govuk_node::s_content_data_api_db_admin
#
# This machine class is used to administer the Content Data API
# PostgreSQL RDS instances.
#
class govuk::node::s_content_data_api_db_admin inherits govuk::node::s_postgresql_db_admin_base {
  # include all PostgreSQL classes that create databases and users
  class { '::govuk::apps::content_data_api::db': }
}
```

...but got a `Must pass apt_mirror_hostname to Class[Govuk::Nodes::S_postgresql_db_admin_base]` error. It [doesn't seem possible to inherit parameterized classes](https://serverfault.com/a/443778). (I did also try [inheriting each parameter explicitly](https://serverfault.com/a/900345)).